CSI: add initial CSI plugin skeleton for DigitalOcean Block Storage

fatih opened the pull request:
This PR adds the necessary dependencies for creating a CSI plugin and a skeleton plugin that implements the following interfaces:

* NodeServer
* IdentityServer
* ControllerServer

Please review the last commit: https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/71/commits/6092e50d2e7d5adf51d3a1efb424e5886ea9cd5a

fatih commented:
I'm still not sure why `dep` pulls the whole kubernetes repo. Maybe our current Gopkl.toml file is not up-to-date. @andrewsykim any ideas?

fatih commented:
@bhcleek but I didn't modified anything in those files. That's why it was confusing for me.

bhcleek commented:
I see. Vendoring kubernetes with dep has some interesting problems. See https://github.com/golang/dep/issues/1339, but that doesn't seem to be a problem here.

Looking at the changes to `Gopkg.lock`, it looks like it just only formatting changes were made to `k8s.io/kubernetes`; the revision remains the same.

New files were also brought in, but they were only tests, commands, and text files. We haven't updated the `Gopkg.toml` to include prune options since `dep` added them, so `dep ensure` doesn't automatically prune packages. Did you run `dep prune` after `dep ensure`? If not, then please do that. If you did, then it's likely just a change to how dep selects which files to vendor.

andrewsykim commented:
hmm, yeah not sure what's up there. I'm okay to merge the changes as they are, we probably needed to update the `k8s.io/kubernetes/pkg/cloudprovider` anyways, AFAICT there hasn't been any big changes there upstream.

andrewsykim commented:
Not sure if the travis CI change is necessary, I see duplicate travis CI status on the PR now

bhcleek commented:
I definitely don't want to merge this as is. Unless I'm reading it wrong, it's vendoring itself: https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/71/files#diff-bd247e83efc3c45ae9e8c47233249f18R41

fatih commented:
I know that there is a problem. All I did was calling “dep ensure”. I’m
going to check it later in my morning.

On Tue, Apr 3, 2018 at 9:00 PM Billie Cleek <notifications@github.com>
wrote:

> I definitely don't want to merge this as is. Unless I'm reading it wrong,
> it's vendoring itself:
> https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/71/files#diff-bd247e83efc3c45ae9e8c47233249f18R41
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/71#issuecomment-378341639>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAayiLICbZTv17dGlU20JTliPUozSSIJks5tk7jTgaJpZM4TFD3P>
> .
>
-- 
Fatih Arslan

fatih commented:
@andrewsykim that can be disabled from the travis CI settings:

<img width="1196" alt="image" src="https://user-images.githubusercontent.com/438920/38292249-b9735912-37eb-11e8-8cab-87ccd0e36f5e.png">

Because otherwise it doesn't build other branches at all. This means that I can't see if my build fails without opening a PR. I've changed the settings now, let's use that way. If it still continues to show two checks, I'll make sure to re add it back. 

@bhcleek @andrewsykim @lxfontes 

I've fixed the dep issue. I've called `dep ensure` followed by a `dep prune` and it removed the self dependency of cloud-controller-manager to itself. I believe this was a the reason Kubernetes was pulled in again as a new dependency.

PTAL

fatih commented:
Closing this in favor of: https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/72 

Which doesn't has the travis change and also doesn't have the history of vendor changes.
